### DISCUSSION TITLE
Next.js 如何在 i18n 场景下处理 404 页面

### DISCUSSION THREAD

**Discussion by giscus** [liruifengv/liruifengv.com#17] [Announcements]
# Next.js 如何在 i18n 场景下处理 404 页面

Next.js 使用 app route，国际化时无法自定义 404 页面。

https://liruifengv.com/posts/nextjs-404-in-i18n/

**Comment by xiguaakako** [comment]
我用的是next-international套件，去翻了issue连作者都说这是next.js固有的问题，即使解决了也很难对404页面做i18n。
没想到在这里找到了答案，太感谢您了。

**Reply by liruifengv** [reply]
不客气，很开心帮助到你

**Comment by Dreamer-Paul** [comment]
> 然后在 app/[lang]/[...not_found]/page.tsx 中使用这个组件。

这种方式的话会导致对应页面直接返回 HTTP 200 状态码，亲测在 14.2.4 版本，可以在你前面执行 `notFound()` 哪个方法下，直接建立 `app/not-found.tsx` 文件，就能够覆盖默认的 404 样式，同时返回 404 状态码，不知道大佬有没有什么看起来更好的办法...
